### ISSUE TITLE
Create Seurat Object with count table and cell information table with .txt file

### ISSUE THREAD

**Issue by azurehk** [satijalab/seurat#6389]
I want to ask how can I create a Seurat Object with .txt file with cell information that are also .txt file? I am so new to Seurat and am not sure how to do it. (https://zenodo.org/record/4387066#.Ytl1uoTMKUk)

library(Seurat)
library(dplyr)

Lac <- read.delim("LacTreatmentSC_Expression.txt.gz")
CellInfo <- read.delim("LacTreatmentSC_CellInfo.txt)

I am not sure how to do the next step... may anyone kindly help me?

**Comment by azurehk** [comment]
Thank you so much
But when I tried to run your code, 
Warning message:
In data.table::fread("LacTreatmentSC.txt", data.table = F) :
  Detected 21826 column names but the data has 21827 columns (i.e. invalid file). Added 1 extra default column name for the first column which is guessed to be row names or an index. Use setnames() afterwards if this guess is not correct, or fix the file write command that created the file to create a valid file.
This is what I got.

**Comment by samuel-marsh** [comment]
Yes that’s just warning message and the next line of code takes care of that moving barcodes to rownames.  It can be ignored.

**Comment by azurehk** [comment]
I tried to ignore the warning and run the code but it failed
 Lac <- data.table::fread("LacTreatmentSC.txt", data.table = F)  %>% column_to_rownames("V1")
Warning message:
In data.table::fread("LacTreatmentSC.txt", data.table = F) :
  Detected 21826 column names but the data has 21827 columns (i.e. invalid file). Added 1 extra default column name for the first column which is guessed to be row names or an index. Use setnames() afterwards if this guess is not correct, or fix the file write command that created the file to create a valid file.
> colnames(Lac) <- gsub(replacement = "-", pattern = ".", x = colnames(Lac), fixed = T)
Error in is.data.frame(x) : object 'Lac' not found
